### PR TITLE
RCPS-108 Prefect System: Make a QueryConstant class

### PR DESCRIPTION
**Pull request by rosellemarithe**

### CODE REVIEW

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java [hunk 1]
```diff
@@ -7,77 +7,79 @@
...
     private static final Logger LOGGER = LoggerFactory.getLogger(CommunityServiceDaoImpl.class);
+    Connection c = ConnectionHelper.getConnection();
     @Override
     public List<CommunityService> getAllCs() {
         List<CommunityService> communityServices = new ArrayList<>();
-        String sql = "SELECT * FROM comm_serv_rendered";
-        try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement statement = con.prepareStatement(sql);
-             ResultSet resultSet = statement.executeQuery()) {
+        try (PreparedStatement stmt = c.prepareStatement(GET_ALL_CS_STATEMENT);
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java:23]: Clean this up. Check with others how to do the try with resource correctly

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java [hunk 2]
```diff
@@ -7,77 +7,79 @@
...
-                    int idNum = rs.getInt("id");
-                    String student_id = rs.getString("student_id");
-                    Timestamp date_rendered = rs.getTimestamp("date_rendered");
-                    int hours_rendered = rs.getInt("hours_rendered");
-                    return new CommunityService(idNum, student_id, date_rendered, hours_rendered);
-                } else {
-                    LOGGER.info("No Community Service found with ID: " + id);
+        @Override
+        public CommunityService getCsById(int id) {
+            try (PreparedStatement stmt = c.prepareStatement(GET_CS_BY_ID_STATEMENT)) {
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java:45]: Clean this up. Check with others how to do the try with resource correctly

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java [hunk 3]
```diff
@@ -7,77 +7,79 @@
...
-        LOGGER.debug("Community Service not found.");
-        return null;
-    }
 
     @Override
     public boolean renderCs(CommunityService cs) throws SQLException {
-        String sql = "UPDATE comm_serv_rendered SET student_id = ?, date_rendered = ?, hours_rendered = ? WHERE id = ?";
-        try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement stmt = con.prepareStatement(sql)) {
+        try ( PreparedStatement stmt = c.prepareStatement(RENDER_CS_STATEMENT);
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java:68]: Clean this up. Check with others how to do the try with resource correctly

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java [hunk 4]
```diff
@@ -9,16 +9,16 @@
...
 
     private static final Logger LOGGER = LoggerFactory.getLogger(OffenseDaoImpl.class);
+    Connection c = ConnectionHelper.getConnection();
 
     @Override
     public Offense getOffenseByID(int id) {
-        String sql = "SELECT * FROM offense WHERE id = ?";
-        try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement stmt = con.prepareStatement(sql)) {
+        try (PreparedStatement stmt = c.prepareStatement(GET_OFFENSE_BY_ID_STATEMENT)) {
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java:21]: Clean this up. Check with others how to do the try with resource correctly

File: src/main/java/com/prefect/office/record/management/appl/facade/prefect/communityservice/impl/CommunityServiceFacadeImpl.java [hunk 5]
```diff
@@ -3,14 +3,12 @@
...
+import com.prefect.office.record.management.data.dao.prefect.communityservice.impl.CommunityServiceDaoImpl;
 
 import java.util.List;
 
 public class CommunityServiceFacadeImpl implements CommunityServiceFacade {
-    private CommunityServiceDao communityServiceDao;
-    public CommunityServiceFacadeImpl(CommunityServiceDao communityServiceDao) {
-        this.communityServiceDao = communityServiceDao;
-    }
+    CommunityServiceDao communityServiceDao = new CommunityServiceDaoImpl();
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/appl/facade/prefect/communityservice/impl/CommunityServiceFacadeImpl.java:11]: where is the access modifier for this?

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java [hunk 6]
```diff
@@ -3,81 +3,89 @@
...
+import java.sql.ResultSet;
 import java.util.ArrayList;
 import java.util.List;
 
+import static com.prefect.office.record.management.data.utils.QueryConstants.*;
+
 public class CommunityServiceDaoImpl implements CommunityServiceDao {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CommunityServiceDaoImpl.class);
+    Connection c = ConnectionHelper.getConnection();
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java:20]: Remove this connection here

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java [hunk 7]
```diff
@@ -3,81 +3,89 @@
...
-            try (ResultSet rs = stmt.executeQuery()) {
-                if (rs.next()) {
-                    int idNum = rs.getInt("id");
-                    String student_id = rs.getString("student_id");
-                    Timestamp date_rendered = rs.getTimestamp("date_rendered");
-                    int hours_rendered = rs.getInt("hours_rendered");
-                    return new CommunityService(idNum, student_id, date_rendered, hours_rendered);
-                } else {
-                    LOGGER.info("No Community Service found with ID: " + id);
+        @Override
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java:47]: why put extra tab space?

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java [hunk 8]
```diff
@@ -3,22 +3,25 @@
...
 import java.sql.*;
 import java.util.ArrayList;
 import java.util.List;
+import static com.prefect.office.record.management.data.utils.QueryConstants.*;
 
 public class OffenseDaoImpl implements OffenseDao {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(OffenseDaoImpl.class);
-
+    Connection c = ConnectionHelper.getConnection();
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java:18]: remove this connection

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/violation/impl/ViolationDaoImpl.java [hunk 9]
```diff
@@ -13,31 +14,41 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import static com.prefect.office.record.management.data.utils.QueryConstants.*;
 public class ViolationDaoImpl implements ViolationDao {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ViolationDaoImpl.class);
+    Connection c = ConnectionHelper.getConnection();
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/violation/impl/ViolationDaoImpl.java:21]: Remove this connection

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java [hunk 10]
```diff
@@ -3,81 +3,89 @@
...
+    public CommunityServiceDaoImpl() {
+    }
     @Override
     public List<CommunityService> getAllCs() {
         List<CommunityService> communityServices = new ArrayList<>();
-        String sql = "SELECT * FROM comm_serv_rendered";
-        try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement statement = con.prepareStatement(sql);
-             ResultSet resultSet = statement.executeQuery()) {
+        try (Connection c = new ConnectionHelper().getConnection()){
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java:26]: how can you create a new instance of a final class?

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java [hunk 11]
```diff
@@ -3,81 +3,89 @@
...
-        LOGGER.debug("Community Service not found.");
-        return null;
-    }
 
     @Override
     public boolean renderCs(CommunityService cs) throws SQLException {
-        String sql = "UPDATE comm_serv_rendered SET student_id = ?, date_rendered = ?, hours_rendered = ? WHERE id = ?";
-        try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement stmt = con.prepareStatement(sql)) {
+        try (Connection c = new ConnectionHelper().getConnection()){
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/communityservice/impl/CommunityServiceDaoImpl.java:73]: how can you create a new instance of a final/static class?

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java [hunk 12]
```diff
@@ -3,22 +3,25 @@
...
+    public OffenseDaoImpl() {
+    }
     @Override
     public Offense getOffenseByID(int id) {
-        String sql = "SELECT * FROM offense WHERE id = ?";
-        try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement stmt = con.prepareStatement(sql)) {
+        try (Connection c = new ConnectionHelper().getConnection()){
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java:23]: how can you create a new instance of a final/static class?

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java [hunk 13]
```diff
@@ -41,16 +44,16 @@ public Offense getOffenseByID(int id) {
 
     @Override
     public boolean updateOffense(Offense offense) {
-        String sql = "UPDATE offense SET violation_id = ?, student_id = ?, offense_date = ? WHERE id = ?";
-        try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement stmt = con.prepareStatement(sql)) {
+        try (Connection c = new ConnectionHelper().getConnection()){
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java:47]: how can you create a new instance of a final/static class?

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java [hunk 14]
```diff
@@ -60,39 +63,40 @@ public boolean updateOffense(Offense offense) {
     @Override
     public List<Offense> getAllOffenses() {
         List<Offense> offenses = new ArrayList<>();
-        String sql = "SELECT * FROM offense";
-        try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement statement = con.prepareStatement(sql);
-             ResultSet resultSet = statement.executeQuery()) {
+        try (Connection c = new ConnectionHelper().getConnection()){
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/offense/impl/OffenseDaoImpl.java:66]: how can you create a new instance of a final/static class?

File: src/main/java/com/prefect/office/record/management/data/dao/prefect/violation/impl/ViolationDaoImpl.java [hunk 15]
```diff
@@ -13,31 +14,41 @@
...
-        try (Connection connection = ConnectionHelper.getConnection();
-             PreparedStatement preparedStatement = connection.prepareStatement("INSERT INTO violation(violation, type, comm_serv_hours) VALUES (?, ?, ?)")) {
-            preparedStatement.setString(1, violation.getViolation());
-            preparedStatement.setString(2, violation.getType());
-            preparedStatement.setInt(3, violation.getCommServHours());
-            preparedStatement.executeUpdate();
-        } catch (SQLException ex) {
-            LOGGER.warn("Error adding violation: " + ex.getMessage());
-            ex.printStackTrace();
+        try (Connection c = new ConnectionHelper().getConnection()){
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/data/dao/prefect/violation/impl/ViolationDaoImpl.java:27]: how can you create a new instance of a final/static class?

**Review by RoCS2024 — CHANGES_REQUESTED** [review]
see ALL comments

**Review by RoCS2024 — CHANGES_REQUESTED** [review]
Revert to QueryConstant class change only., We will just create a separate ticket for the rest.

**Comment by rosellemarithe** [comment]
sir I created a new branch and will close this nalang po

**Pull request closed without merging**